x-pack/plugins/security/public/management/roles/edit_role/edit_role_page.tsx

```diff
@@ -302,6 +302,8 @@ export const EditRolePage: FunctionComponent<Props> = ({
   // eventually enable validation after the first time user tries to save a role.
   const { current: validator } = useRef(new RoleValidator({ shouldValidate: false }));
   const [formError, setFormError] = useState<RoleValidationResult | null>(null);
+  const [creatingRoleAlreadyExists, setCreatingRoleAlreadyExists] = useState<boolean>(false);
```

Comment:
Nice job on the UI!
It's a little weird that even though you're catching 404 errors for the Get Role API call, it still shows up as an error in the browser console. I guess it's unavoidable unless we wanted to create an entirely separate route that responds with a 200 status code no matter what, and I think that's overkill.

I thought of two more things that would be nice to have:

1. If the role name is already taken, we can disable the "Create role" button
2. If we encounter the race condition where the role name _isn't_ taken, but then the Create API call results in a 409 error, we can show the inline validation error

Let me know what you think:

```diff
diff --git a/x-pack/plugins/security/public/management/roles/edit_role/edit_role_page.tsx b/x-pack/plugins/security/public/management/roles/edit_role/edit_role_page.tsx
index 7ae2848f00d..34d01b14942 100644
--- a/x-pack/plugins/security/public/management/roles/edit_role/edit_role_page.tsx
+++ b/x-pack/plugins/security/public/management/roles/edit_role/edit_role_page.tsx
@@ -518,7 +518,7 @@ export const EditRolePage: FunctionComponent<Props> = ({
         data-test-subj={`roleFormSaveButton`}
         fill
         onClick={saveRole}
-        disabled={isRoleReserved}
+        disabled={isRoleReserved || creatingRoleAlreadyExists}
       >
         {saveText}
       </EuiButton>
@@ -548,6 +548,11 @@ export const EditRolePage: FunctionComponent<Props> = ({
       try {
         await rolesAPIClient.saveRole({ role, createOnly: !isEditingExistingRole });
       } catch (error) {
+        if (error?.body?.statusCode === 409) {
+          setCreatingRoleAlreadyExists(true);
+          window.scroll({ top: 0, behavior: 'smooth' });
+          return;
+        }
         notifications.toasts.addDanger(
           error?.body?.message ??
             i18n.translate('xpack.security.management.editRole.errorSavingRoleError', {
```

Reply:
I like this suggestion a lot. I think is the behavior I would ultimately expect from a form. Thanks, Joe!